app/controllers/application_controller.rb

```diff
@@ -1,2 +1,13 @@
 class ApplicationController < ActionController::Base
+  before_action :authenticate_user!
+  before_action :set_current_user
+  before_action :store_previous_page, only: [:new]
+
+  def set_current_user
+    @user = current_user if user_signed_in?
+  end
```

Comment:
- [Optional] Dear @redmarverick, for the buttons I suggest you follow the design guideline closely, as you can see on the original design the button has the name of the item to be added instead of just `+`.

Reply:
The original design does not show any name for the buttons, Im pretty sure I said on the video the function of the Plus button on categories page and transactions page
https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=&

Reply:
- [ ] Dear @redmarverick, this is the original design for the button 
![Screen Shot 2023-09-16 at 12 08 01 PM](https://github.com/redmarverick/FourEconomy/assets/69077061/87728963-0ae9-443c-9d3e-c99e25464948)
**Your design is great, I just wanna mention you could have followed the guideline better**